engine/apps/slack/migrations/0007_migrate_slackmessage_channel_id.py

```diff
@@ -0,0 +1,78 @@
+# Generated by Django 4.2.16 on 2024-11-01 10:58
+import logging
+
+from django.conf import settings
+from django.db import migrations
+import django_migration_linter as linter
+
+logger = logging.getLogger(__name__)
+
+def populate_slack_channel(apps, schema_editor):
+    SlackMessage = apps.get_model("slack", "SlackMessage")
+    SlackChannel = apps.get_model("slack", "SlackChannel")
+
+    logger.info("Starting migration to populate channel field.")
+
+    db_type = getattr(settings, "DATABASE_TYPE", None)
+
+    if db_type == settings.DATABASE_TYPES.POSTGRESQL:
+        sql = f"""
+        UPDATE {SlackMessage._meta.db_table} AS sm
+        SET channel_id = sc.id
+        FROM {SlackChannel._meta.db_table} AS sc
+        WHERE sc.slack_id = sm._channel_id
+          AND sc.slack_team_identity_id = sm.slack_team_identity
+          AND sm._channel_id IS NOT NULL
+          AND sm.slack_team_identity IS NOT NULL;
+        """
+    elif db_type == settings.DATABASE_TYPES.SQLITE3:
+        sql = f"""
+        UPDATE {SlackMessage._meta.db_table}
+        SET channel_id = (
+            SELECT sc.id
+            FROM {SlackChannel._meta.db_table} AS sc
+            WHERE sc.slack_id = {SlackMessage._meta.db_table}._channel_id
+              AND sc.slack_team_identity_id = {SlackMessage._meta.db_table}.slack_team_identity
+        )
+        WHERE _channel_id IS NOT NULL
+          AND slack_team_identity IS NOT NULL
+          AND EXISTS (
+              SELECT 1
+              FROM {SlackChannel._meta.db_table} AS sc
+              WHERE sc.slack_id = {SlackMessage._meta.db_table}._channel_id
+                AND sc.slack_team_identity_id = {SlackMessage._meta.db_table}.slack_team_identity
+          );
+        """
+    elif db_type == settings.DATABASE_TYPES.MYSQL:
+        sql = f"""
+        UPDATE
+            {SlackMessage._meta.db_table} AS sm
+        JOIN {SlackChannel._meta.db_table} AS sc
+            ON sc.slack_id = sm._channel_id
+            AND sc.slack_team_identity_id = sm.slack_team_identity
+        SET
+            sm.channel_id = sc.id
+        WHERE
+            sm._channel_id IS NOT NULL
+            AND sm.slack_team_identity IS NOT NULL;
+        """
```

Comment:
main change. Added queries for sqlite/postgres here to avoid https://github.com/grafana/oncall/issues/5244#issuecomment-2493688544

Reply:
TODO: I'm planning on running this SQL statement against a prod clone just to get an idea of performance before merging this PR